From d799c0372e6bde0f6b8786bb1705de46115b05b6 Mon Sep 17 00:00:00 2001
From: anuj batham <anuj.batham@nxp.com>
Date: Fri, 27 Apr 2018 14:38:09 +0530
Subject: [PATCH 16/22] staging: fsl_ppfe/eth: HW parse results for DPDK

HW Parse results are included in the packet headroom.
Length and Offset calculation now accommodates parse info size.

Signed-off-by: Archana Madhavan <archana.madhavan@nxp.com>

[ Srinidhi Rao : Ported this patch to photon linux from
  qoriq-components linux 'linux-4.14-nxp'
  (https://source.codeaurora.org/external/qoriq/qoriq-components/linux)
]

Signed-off-by: srinidhira0 <srinidhir@vmware.com>
---
 drivers/staging/fsl_ppfe/pfe_hif_lib.c | 7 +++++--
 drivers/staging/fsl_ppfe/pfe_hif_lib.h | 1 +
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl_ppfe/pfe_hif_lib.c b/drivers/staging/fsl_ppfe/pfe_hif_lib.c
index eeab4ff3b6b8..16fcf2daeef8 100644
--- a/drivers/staging/fsl_ppfe/pfe_hif_lib.c
+++ b/drivers/staging/fsl_ppfe/pfe_hif_lib.c
@@ -435,6 +435,7 @@ void *hif_lib_receive_pkt(struct hif_client_s *client, int qno, int *len, int
 			u16 size = *rx_ctrl >> HIF_CTRL_RX_OFFSET_OFST;
 
 			if (size) {
+				size += PFE_PARSE_INFO_SIZE;
 				*len = CL_DESC_BUF_LEN(desc->ctrl) -
 						PFE_PKT_HEADER_SZ - size;
 				*ofst = pfe_pkt_headroom + PFE_PKT_HEADER_SZ
@@ -442,8 +443,10 @@ void *hif_lib_receive_pkt(struct hif_client_s *client, int qno, int *len, int
 				*priv_data = desc->data + PFE_PKT_HEADER_SZ;
 			} else {
 				*len = CL_DESC_BUF_LEN(desc->ctrl) -
-						PFE_PKT_HEADER_SZ;
-				*ofst = pfe_pkt_headroom + PFE_PKT_HEADER_SZ;
+				       PFE_PKT_HEADER_SZ - PFE_PARSE_INFO_SIZE;
+				*ofst = pfe_pkt_headroom
+					+ PFE_PKT_HEADER_SZ
+					+ PFE_PARSE_INFO_SIZE;
 				*priv_data = NULL;
 			}
 
diff --git a/drivers/staging/fsl_ppfe/pfe_hif_lib.h b/drivers/staging/fsl_ppfe/pfe_hif_lib.h
index 08031f1f3117..c8334a6b6a13 100644
--- a/drivers/staging/fsl_ppfe/pfe_hif_lib.h
+++ b/drivers/staging/fsl_ppfe/pfe_hif_lib.h
@@ -23,6 +23,7 @@
 
 #define HIF_CL_REQ_TIMEOUT	10
 #define GFP_DMA_PFE 0
+#define PFE_PARSE_INFO_SIZE	16
 
 enum {
 	REQUEST_CL_REGISTER = 0,
-- 
2.14.2

